packages/headless/src/features/facets/automatic-facet-set/automatic-facet-set-slice.ts

```diff
@@ -48,6 +51,39 @@ export const automaticFacetSetReducer = createReducer(
         for (const value of facet.values) {
           value.state = 'idle';
         }
+      })
+      .addCase(restoreSearchParameters, (state, action) => {
+        const af = action.payload.af ?? {};
+        const fields = Object.keys(af);
+
+        for (const field of fields) {
+          const response = buildTemporaryAutomaticFacetResponse(field);
+          const values = af[field].map((value) =>
+            buildTemporarySelectedFacetValue(value)
+          );
+          response.values.push(...values);
+
+          state.facets[field] = response;
```

Comment:
so in this case the `field` is present in both the key and value? Is this dup necessary?

Reply:
Yea, the field is used as a key. I followed the same pattern as the other facet sets and I think it is fine to use it this way.